pkg/csi-common/driver_test.go

```diff
@@ -0,0 +1,165 @@
+package csicommon
+
+import (
+	"testing"
+
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	fakeDriverName = "fake"
+	fakeNodeID     = "fakeNodeID"
+)
+
+var (
+	fakeVersion           = csi.Version{Major: 5, Minor: 2, Patch: 0}
+	fakeVersionsSupported = []*csi.Version{
+		&csi.Version{
+			Major: 4, Minor: 0, Patch: 0,
+		},
+		&csi.Version{
+			Major: 4, Minor: 1, Patch: 0,
+		},
+	}
+)
+
+func NewFakeDriver() *CSIDriver {
+	fakeVersion = csi.Version{Major: 5, Minor: 2, Patch: 0}
+	fakeVersionsSupported = []*csi.Version{
+		&csi.Version{
+			Major: 4, Minor: 0, Patch: 0,
+		},
+		&csi.Version{
+			Major: 4, Minor: 1, Patch: 0,
+		},
+	}
+
+	driver := NewCSIDriver(fakeDriverName, &fakeVersion, fakeVersionsSupported, fakeNodeID)
+
+	return driver
+}
+
+func TestNewFakeDriver(t *testing.T) {
+	// Test New fake driver with invalid arguments.
+	d := NewCSIDriver("", &fakeVersion, fakeVersionsSupported, fakeNodeID)
+	assert.Nil(t, d)
+}
+
+func TestCheckVersion(t *testing.T) {
+
+	driver := NewFakeDriver()
+
+	// Exact version
+	v := csi.Version{
+		Major: 5,
+		Minor: 1,
+		Patch: 0,
+	}
+	err := driver.CheckVersion(&v)
+	assert.NoError(t, err)
+
+	//Supported version
+	v = csi.Version{
+		Major: 4,
+		Minor: 0,
+		Patch: 0,
+	}
+	err = driver.CheckVersion(&v)
+	assert.NoError(t, err)
+
+	// Unsupported version
+	v = csi.Version{
+		Major: 6,
+		Minor: 0,
+		Patch: 0,
+	}
+	err = driver.CheckVersion(&v)
+	s, ok := status.FromError(err)
+	assert.True(t, ok)
+	assert.Equal(t, s.Code(), codes.InvalidArgument)
+
+	// Supported minor version
+	v = csi.Version{
+		Major: 5,
+		Minor: 1,
+		Patch: 0,
+	}
+	err = driver.CheckVersion(&v)
+	assert.NoError(t, err)
+
+	// Unsupported minor version
+	v = csi.Version{
+		Major: 5,
+		Minor: 3,
+		Patch: 0,
+	}
+	err = driver.CheckVersion(&v)
+	s, ok = status.FromError(err)
+	assert.True(t, ok)
+	assert.Equal(t, s.Code(), codes.InvalidArgument)
+}
+
+func TestGetVolumeCapabilityAccessModes(t *testing.T) {
```

Comment:
what about access type (mount, block)?

Reply:
That is VolumeCapability. As of now, we don't expose VolumeCapability in the driver base. Let me see if I can abstract it out too. Will be next PR though :)